app/models/ems_refresh/save_inventory_container.rb

```diff
@@ -1,436 +1,164 @@
-
-    ems.container_nodes(true)
-    deletes = if target.kind_of?(ExtManagementSystem)
-                ems.container_nodes.dup
-              else
-                []
-              end
-
-    save_inventory_multi(ems.container_nodes, hashes, deletes, [:ems_ref],
-                         [:labels, :computer_system, :container_conditions], [:namespace])
-    store_ids_for_new_records(ems.container_nodes, hashes, :ems_ref)
-  end
-
-  def save_computer_system_inventory(container_node, hash, _target = nil)
-    save_inventory_single(:computer_system, container_node, hash, [:hardware, :operating_system])
-  end
-
-  def save_container_replicators_inventory(ems, hashes, target = nil)
-    return if hashes.nil?
-    target = ems if target.nil?
-
-    ems.container_replicators(true)
-    deletes = if target.kind_of?(ExtManagementSystem)
-                ems.container_replicators.dup
-              else
-                []
-              end
-
-    hashes.each do |h|
-      h[:container_project_id] = h.fetch_path(:project, :id)
-    end
-
-    save_inventory_multi(ems.container_replicators, hashes, deletes, [:ems_ref],
-                         [:labels, :selector_parts], [:project, :namespace])
-    store_ids_for_new_records(ems.container_replicators, hashes, :ems_ref)
-  end
-
-  def save_container_services_inventory(ems, hashes, target = nil)
-    return if hashes.nil?
-    target = ems if target.nil?
-
-    ems.container_services(true)
-    deletes = if target.kind_of?(ExtManagementSystem)
-                ems.container_services.dup
-              else
-                []
-              end
-
-    hashes.each do |h|
-      h[:container_group_ids] = h[:container_groups].map { |x| x[:id] }
-      h[:container_project_id] = h.fetch_path(:project, :id)
-    end
-
-    save_inventory_multi(ems.container_services, hashes, deletes, [:ems_ref],
-                         [:labels, :selector_parts, :container_service_port_configs],
-                         [:container_groups, :project, :namespace])
-
-    store_ids_for_new_records(ems.container_services, hashes, :ems_ref)
-  end
-
-  def save_container_groups_inventory(ems, hashes, target = nil)
-    return if hashes.nil?
-    target = ems if target.nil?
-
-    ems.container_groups(true)
-    deletes = if target.kind_of?(ExtManagementSystem)
-                ems.container_groups.dup
-              else
-                []
-              end
-
-    hashes.each do |h|
-      h[:container_node_id] = h.fetch_path(:container_node, :id)
-      h[:container_replicator_id] = h.fetch_path(:container_replicator, :id)
-      h[:container_project_id] = h.fetch_path(:project, :id)
-      h[:container_build_pod_id] = ems.container_build_pods.find_by_name(
-        h[:build_pod_name]).try(:id)
-    end
-
-    save_inventory_multi(ems.container_groups, hashes, deletes, [:ems_ref],
-                         [:container_definitions, :containers, :labels, :node_selector_parts, :container_conditions,
```

Comment:
I suspect :containers here might have no meaning and should be removed. containers are saved through container_definition.
@zeari @dkorn @simon3z any idea?